src/Controllers/WishlistController.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Rapidez\MultipleWishlist\Controllers;
+
+use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
+use Illuminate\Foundation\Bus\DispatchesJobs;
+use Illuminate\Foundation\Validation\ValidatesRequests;
+use Illuminate\Support\Collection;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Facades\DB;
+use Rapidez\Core\Models\Product;
+use Rapidez\MultipleWishlist\Models\MagentoWishlist;
+use Rapidez\MultipleWishlist\Models\Wishlist;
+
+class WishlistController extends Controller
+{
+    use AuthorizesRequests;
+    use DispatchesJobs;
+    use ValidatesRequests;
+
+    public function index(Request $request): Collection
+    {
+        // There has to be a way to do this part more succinctly
+        return Wishlist::select(['jb_wishlist.*', DB::raw('COUNT(jb_wishlist_item.id) AS item_count')])
+            ->where('customer_id', $request->userId)
+            ->leftJoin('jb_wishlist_item', 'jb_wishlist_item.wishlist_id', '=', 'jb_wishlist.id')
+            ->groupBy('jb_wishlist.id')
+            ->get();
+    }
+
+    public function show(Request $request, $id): mixed
+    {
+        $wl = Wishlist::where('id', $id)->where('customer_id', $request->userId)->first();
+        if (!$wl) {
+            return 'Wishlist not found';
+        }
+
+        return [$wl, $wl->items()->get()];
+    }
```

Comment:
This could become something like this:
```
public function show(Request $request, Wishlist $wishlist)
{
    return $wishlist->load('items');
}
```
Where the authentication can be handled with policies: https://laravel.com/docs/master/authorization#creating-policies